.changeset/dry-suns-smash.md

```diff
@@ -0,0 +1,5 @@
+---
+'playroom': patch
+---
+
+Highlight the correct error location when code has syntax errors
```

Comment:
Feel like this is a bit misleading as it's fixing an unreleased bug caused by the sucrase migration. I guess what I don't want to happen is a consumer to think that there was something broken in a previous release,

Reply:
The issue was with `.trim()` applied to the code before transform https://github.com/seek-oss/playroom/blob/c4b639ced4f1abbb788e94ec8e8dd8eda4ff9dbe/src/utils/compileJsx.ts#L7

It produced error markers at the wrong location:

<img width="457" alt="wrong error location" src="https://github.com/seek-oss/playroom/assets/3297808/aebeba94-32c8-467c-9761-4533b41603ea">

<img width="457" alt="correct error location" src="https://github.com/seek-oss/playroom/assets/3297808/d7b9f35d-8ef8-481b-a2bd-7c000a62eef5">

Reply:
From https://github.com/seek-oss/playroom/actions/runs/6804897208:

https://playroom--9ad9283763070a49c3755107edb71ca95d2b41b3.surge.sh/?code=N4Igxg9gJgpiBcIA6A7V6UGsaYAQB4AxCCXSAGwgCcBeJEKmKegPmNIEZ8B6dl1bHiIkyESlXjw6IAEbkAhmEyt2uAEw8+AnAVUVq0xsxBsRAZk0l+WHcNL7a9IypEAWSxGsgAvkA

Reply:
Oh, so this was unrelated to the sucrase change?

Reply:
[Yes](https://seek-oss.github.io/braid-design-system/playroom/#?code=N4Igxg9gJgpiBcIA6A7VqA8BBMkBOUAlhCgATzxEBuhseAzgLzABmAhgDb0wC%2BAfKlKlsuCAWIoAkgBcYAW1Ic2AIxgdGSEDPmkAjJoFkhwgApKwMABYQOdUpZiEA5penNdABg89SAekNCGL44%2BEQk2nIBwiFiYVKyCkqq6poRpABMBoLGGGZsFta2MHj2ji5uwJ7eflFBMeLhCbX1cWlJahpaCaQAzFlGgXkFNnYOzq7uXj7%2B2XWiDfHyhnOhEoYgADQg0g5yMPQIANogbAAO%2BZsg3DAA1gBSEMoHALpbAO60Owfwhz3pHhtdAAWLzPHhAA)